### PR TITLE
Defered in pools

### PR DESCRIPTION
**Pull request by tim-habitat**
attempt at adding https://github.com/DrFaust92/terraform-provider-airflow/issues/35

### CODE REVIEW

**Comment by tim-habitat** [comment]
hey @DrFaust92 what do you think of adding this toggle to the `pools` ? :)

File: internal/provider/resource_pool.go [hunk 1]
```diff
@@ -53,19 +57,24 @@ func resourcePoolCreate(ctx context.Context, d *schema.ResourceData, m interface
...
 
-	_, _, err := varApi.PostPool(pcfg.AuthContext).Pool(pool).Execute()
+	_, _, err := varApi.PostPool(pcfg.AuthContext).Pool(airflowPool).Execute()
 	if err != nil {
 		return diag.Errorf("failed to create pool `%s` from Airflow: %s", name, err)
 	}
 	d.SetId(name)
 
+	// Manually manage include_deferred in Terraform state
+	d.Set("include_deferred", includeDeferred)
```
Comment (DrFaust92) [inline internal/provider/resource_pool.go:76]: Why? Should be set on read

File: internal/provider/resource_pool.go [hunk 2]
```diff
@@ -89,6 +98,10 @@ func resourcePoolRead(ctx context.Context, d *schema.ResourceData, m interface{}
 	d.Set("open_slots", pool.OpenSlots)
 	d.Set("used_slots", pool.UsedSlots)
 
+	// Manually manage include_deferred in Terraform state
+	includeDeferred := d.Get("include_deferred").(bool)
```
Comment (DrFaust92) [inline internal/provider/resource_pool.go:102]: Extraneous, let's align with other set examples on that are above this do d.Get(...) inline in the set call

File: internal/provider/resource_pool.go [hunk 3]
```diff
@@ -98,17 +111,21 @@ func resourcePoolUpdate(ctx context.Context, d *schema.ResourceData, m interface
...
 	}
 
-	_, _, err := client.PoolApi.PatchPool(pcfg.AuthContext, name).Pool(pool).Execute()
+	_, _, err := client.PoolApi.PatchPool(pcfg.AuthContext, name).Pool(airflowPool).Execute()
 	if err != nil {
 		return diag.Errorf("failed to update pool `%s` from Airflow: %s", name, err)
 	}
 
+	// Manually manage include_deferred in Terraform state
+	d.Set("include_deferred", includeDeferred)
```
Comment (DrFaust92) [inline internal/provider/resource_pool.go:127]: Same as above in create func

File: internal/provider/resource_pool_test.go [hunk 4]
```diff
@@ -19,11 +19,12 @@ func TestAccAirflowPool_basic(t *testing.T) {
 		CheckDestroy: testAccCheckAirflowPoolCheckDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccAirflowPoolConfigBasic(rName, 2),
+				Config: testAccAirflowPoolConfigBasic(rName, 2, true),
```
Comment (DrFaust92) [inline internal/provider/resource_pool_test.go:22]: Let's not change the existing test and add a new one

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
See comments, plus missing documentation.

**Comment by DrFaust92** [comment]
Ok, I see the problem. This won't work (yet) as the provider is using an older version of the go sdk (2.5.0) and needs 2.7.0 to support this parameter. So we cannot merge this yet, and as far as I recall the go sdk hasn't been built yet for 2.7.0 and requires action upstream.

address the comments for now and I'll take a look at updating the sdk version. 

I'm currently on vacation so this won't happen until at least 2 weeks from now. I'll try to get to this asap

**Comment by kmehkeri** [comment]
Any chance to have this? I would be happy to contribute if needed, just having the baseline would be helpful, is it possible to restore @tim-habitat?

**Comment by tim-habitat** [comment]
hey @kmehkeri sure, here i reinstated it and rebased off the curreent latest master of this repo.

**Comment by kmehkeri** [comment]
Uh... I went into this and only now realized what exactly DrFaust's earlier comment was about (dependency on PRs in [Go Airflow client](https://github.com/apache/airflow-client-go/pull/43) and [Airflow](https://github.com/apache/airflow/pull/34805) itself). So indeed this needs to wait :/

**Comment by brokenjacobs** [comment]
Any updates @DrFaust92 ? I tried implementing this and ran into the same issue with the upstream airflow-go sdk... Since it's just generated from the openAPI spec seems we could just make another project for it if they won't maintain it...

**Comment by DrFaust92** [comment]
available in 0.18.1

**Pull request closed without merging**